datalad/customremotes/__init__.py

```diff
@@ -14,11 +14,10 @@
 
 __all__ = ['RemoteError', 'SpecialRemote']
 
-from annexremote import (
-    ProtocolError,
-    SpecialRemote as _SpecialRemote,
-    RemoteError as _RemoteError,
-)
+from annexremote import ProtocolError
+from annexremote import RemoteError as _RemoteError
+from annexremote import SpecialRemote as _SpecialRemote
```

Comment:
haven't found anything along that line in the doc.

Reply:
I found it!

```
  --ca, --combine-as    Combines as imports on the same line.
```
or as config 
```
❯ isort --show-config | grep 'combine_as'
    "combine_as_imports": false,
            "combine_as_imports": false,
```
so with
```patch
❯ git diff
diff --git a/pyproject.toml b/pyproject.toml
index 06d2cf385..b5a1ae884 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -6,3 +6,4 @@ requires = ["packaging", "setuptools>=40.8.0", "wheel"]
 force_grid_wrap = 2
 include_trailing_comma = true
 multi_line_output = 3
+combine_as_imports = true
```

we would get
```
❯ isort  datalad/customremotes/__init__.py
Fixing /home/yoh/proj/datalad/datalad-maint/datalad/customremotes/__init__.py

❯ git diff
diff --git a/datalad/customremotes/__init__.py b/datalad/customremotes/__init__.py
index f232d5b87..e01644ac5 100644
--- a/datalad/customremotes/__init__.py
+++ b/datalad/customremotes/__init__.py
@@ -16,9 +16,10 @@ __all__ = ['RemoteError', 'SpecialRemote']
 
 from annexremote import (
     ProtocolError,
-    SpecialRemote as _SpecialRemote,
     RemoteError as _RemoteError,
+    SpecialRemote as _SpecialRemote,
 )
+
 from datalad.support.exceptions import format_exception_with_cause
 from datalad.ui import ui
 
```

I was ready to push the changes for that but can't -- I guess you didn't allow maintainers to contribute to your PR?

FWIW -- I just did above config adjustment and then

```
datalad run -m "Rerun pre-commit (for isort) after allowing to combine as imports" 'pre-commit run --all || :'
```

Reply:
but may be we should forget about this! 

@mih expressed his desire to be as he has it in datasalad. There he uses `hatch fmt`. I checked and `hatch fmt` does similar non-combined as imports.
It does it a little differently though!

<details>
<summary>here is a diff from running  precommit run -a AFTER  hatch fmt which tunes it its way</summary> 

```diff
diff --git a/_datalad_build_support/setup.py b/_datalad_build_support/setup.py
index 7076051bc..1c8e78abc 100644
--- a/_datalad_build_support/setup.py
+++ b/_datalad_build_support/setup.py
@@ -8,9 +8,10 @@
 
 import datetime
 import os
-from os.path import dirname, splitext
+from os.path import dirname
 from os.path import join as opj
 from os.path import sep as pathsep
+from os.path import splitext
 
 import setuptools
 from packaging.version import Version
diff --git a/benchmarks/repo.py b/benchmarks/repo.py
index b20688ff9..679b9b404 100644
--- a/benchmarks/repo.py
+++ b/benchmarks/repo.py
@@ -7,9 +7,7 @@
 # ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
 """Benchmarks of the basic repos (Git/Annex) functionality"""
 
-from .common import (
-    SampleSuperDatasetBenchmarks,
-)
+from .common import SampleSuperDatasetBenchmarks
 
 
 # TODO: probably SampleSuperDatasetBenchmarks is not the best for these benchmarks
diff --git a/datalad/cli/helpers.py b/datalad/cli/helpers.py
index b485ba186..bdee9ee48 100644
--- a/datalad/cli/helpers.py
+++ b/datalad/cli/helpers.py
@@ -22,7 +22,6 @@ from textwrap import wrap
 from platformdirs import AppDirs
 
 from datalad import __version__
-
 # delay?
 from datalad.support.exceptions import CapturedException
 from datalad.ui.utils import get_console_width
diff --git a/datalad/cmd.py b/datalad/cmd.py
index 21a5702fe..4e157a326 100644
--- a/datalad/cmd.py
+++ b/datalad/cmd.py
@@ -31,28 +31,19 @@ from typing import (
     Tuple,
     Union,
 )
-from weakref import (
-    WeakValueDictionary,
-)
+from weakref import WeakValueDictionary
 
 from datalad import cfg
-
 # start of legacy import block
 # to avoid breakage of code written before datalad.runner
-from datalad.runner.coreprotocols import (
-    StdOutErrCapture,
-)
-from datalad.runner.gitrunner import (
-    GitRunnerBase,
-)
+from datalad.runner.coreprotocols import StdOutErrCapture
+from datalad.runner.gitrunner import GitRunnerBase
 from datalad.runner.nonasyncrunner import (
     STDERR_FILENO,
     STDOUT_FILENO,
     _ResultGenerator,
 )
-from datalad.runner.protocol import (
-    GeneratorMixIn,
-)
+from datalad.runner.protocol import GeneratorMixIn
 from datalad.runner.runner import WitlessRunner
 from datalad.runner.utils import LineSplitter
 from datalad.support.exceptions import CommandError
diff --git a/datalad/cmdline/main.py b/datalad/cmdline/main.py
index 678a08dae..62cdd336d 100644
--- a/datalad/cmdline/main.py
+++ b/datalad/cmdline/main.py
@@ -16,4 +16,3 @@ import warnings
 warnings.warn("datalad.cmdline.main was replaced by datalad.cli.main in "
               "datalad 0.16. Please update and reinstall extensions.",
               DeprecationWarning)
-
diff --git a/datalad/core/distributed/clone_ria.py b/datalad/core/distributed/clone_ria.py
index f242db192..8b1a4ff60 100644
--- a/datalad/core/distributed/clone_ria.py
+++ b/datalad/core/distributed/clone_ria.py
@@ -8,7 +8,6 @@ from typing import Dict
 from datalad.distribution.dataset import Dataset
 
 from . import clone as mod_clone
-
 # For now kept in clone_utils, to avoid circular import (see datalad-next)
 from .clone_utils import (
     postclone_preannex_cfg_ria,
diff --git a/datalad/core/local/tests/test_diff.py b/datalad/core/local/tests/test_diff.py
index c720d8456..6e2b5fecc 100644
--- a/datalad/core/local/tests/test_diff.py
+++ b/datalad/core/local/tests/test_diff.py
@@ -16,9 +16,7 @@ import os.path as op
 from unittest.mock import patch
 
 import datalad.utils as ut
-from datalad.api import (
-    diff,
-)
+from datalad.api import diff
 from datalad.cmd import (
     GitWitlessRunner,
     StdOutCapture,
diff --git a/datalad/distributed/create_sibling_gitlab.py b/datalad/distributed/create_sibling_gitlab.py
index fc20fc8bc..415d3b47f 100644
--- a/datalad/distributed/create_sibling_gitlab.py
+++ b/datalad/distributed/create_sibling_gitlab.py
@@ -23,7 +23,6 @@ from ..distribution.dataset import (
     require_dataset,
     resolve_path,
 )
-
 # bound methods
 from ..interface.base import (
     Interface,
diff --git a/datalad/distributed/ora_remote.py b/datalad/distributed/ora_remote.py
index 70a21a066..1d251799a 100644
--- a/datalad/distributed/ora_remote.py
+++ b/datalad/distributed/ora_remote.py
@@ -34,9 +34,7 @@ from datalad.support.exceptions import (
     DownloadError,
 )
 from datalad.support.network import url_path2local_path
-from datalad.utils import (
-    ensure_write_permission,
-)
+from datalad.utils import ensure_write_permission
 
 lgr = logging.getLogger('datalad.customremotes.ria_remote')
 
diff --git a/datalad/distributed/tests/test_export_to_figshare.py b/datalad/distributed/tests/test_export_to_figshare.py
index 60f47b229..a620dd30a 100644
--- a/datalad/distributed/tests/test_export_to_figshare.py
+++ b/datalad/distributed/tests/test_export_to_figshare.py
@@ -8,9 +8,7 @@
 # ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
 """Test export_to_figshare"""
 
-from datalad.api import (
-    Dataset,
-)
+from datalad.api import Dataset
 from datalad.support import path as op
 from datalad.support.gitrepo import GitRepo
 from datalad.tests.utils_pytest import (
diff --git a/datalad/distributed/tests/test_ria_basics.py b/datalad/distributed/tests/test_ria_basics.py
index 153407eb9..a4748851c 100644
--- a/datalad/distributed/tests/test_ria_basics.py
+++ b/datalad/distributed/tests/test_ria_basics.py
@@ -10,9 +10,7 @@
 import logging
 import stat
 
-from datalad.api import (
-    Dataset,
-)
+from datalad.api import Dataset
 from datalad.cmd import NoCapture
 from datalad.customremotes.ria_utils import (
     create_ds_in_store,
diff --git a/datalad/distributed/tests/test_ria_git_remote.py b/datalad/distributed/tests/test_ria_git_remote.py
index d18fd46f7..df1162bcd 100644
--- a/datalad/distributed/tests/test_ria_git_remote.py
+++ b/datalad/distributed/tests/test_ria_git_remote.py
@@ -10,9 +10,7 @@
 
 import subprocess
 
-from datalad.api import (
-    Dataset,
-)
+from datalad.api import Dataset
 from datalad.customremotes.ria_utils import (
     create_ds_in_store,
     create_store,
diff --git a/datalad/distribution/create_sibling.py b/datalad/distribution/create_sibling.py
index f31db1e6c..84d4e1784 100644
--- a/datalad/distribution/create_sibling.py
+++ b/datalad/distribution/create_sibling.py
@@ -14,12 +14,12 @@ __docformat__ = 'restructuredtext'
 import logging
 import os
 import shlex
+from os.path import curdir
+from os.path import join as opj
 from os.path import (
-    curdir,
     normpath,
     relpath,
 )
-from os.path import join as opj
 
 from looseversion import LooseVersion
 
@@ -86,9 +86,9 @@ from datalad.utils import (
     ensure_list,
     make_tempfile,
     on_windows,
-    slash_join,
 )
 from datalad.utils import quote_cmdlinearg as sh_quote
+from datalad.utils import slash_join
 
 lgr = logging.getLogger('datalad.distribution.create_sibling')
 # Window's own mkdir command creates intermediate directories by default
diff --git a/datalad/distribution/create_sibling_github.py b/datalad/distribution/create_sibling_github.py
index 200dfa17b..9ed39bf72 100644
--- a/datalad/distribution/create_sibling_github.py
+++ b/datalad/distribution/create_sibling_github.py
@@ -14,4 +14,3 @@ warnings.warn(
     "CreateSiblingGithub has been moved to "
     "datalad.distributed.create_sibling_github. Please adjust the import.",
     DeprecationWarning)
-
diff --git a/datalad/distribution/dataset.py b/datalad/distribution/dataset.py
index 19769b974..34d54d778 100644
--- a/datalad/distribution/dataset.py
+++ b/datalad/distribution/dataset.py
@@ -14,10 +14,12 @@ from functools import wraps
 from os.path import (
     curdir,
     exists,
+)
+from os.path import join as opj
+from os.path import (
     normpath,
     pardir,
 )
-from os.path import join as opj
 from weakref import WeakValueDictionary
 
 import datalad.utils as ut
@@ -30,17 +32,17 @@ from datalad.dataset.repo import (
 from datalad.support import path as op
 from datalad.support.annexrepo import AnnexRepo
 from datalad.support.constraints import Constraint
-
 # DueCredit
 from datalad.support.due import due
 from datalad.support.due_utils import duecredit_dataset
 from datalad.support.exceptions import NoDatasetFound
 from datalad.support.gitrepo import GitRepo
+from datalad.utils import \
+    get_dataset_root  # TODO remove after a while, when external consumers have adjusted; to use get_dataset_root()
 from datalad.utils import (
     Path,
     PurePath,
     ensure_list,
-    get_dataset_root,  # TODO remove after a while, when external consumers have adjusted; to use get_dataset_root()
     getpwd,
     optional_args,
 )
diff --git a/datalad/distribution/tests/test_dataset.py b/datalad/distribution/tests/test_dataset.py
index 7e4d24b12..5f31954a2 100644
--- a/datalad/distribution/tests/test_dataset.py
+++ b/datalad/distribution/tests/test_dataset.py
@@ -11,12 +11,12 @@
 
 import os
 import os.path as op
+from os.path import abspath
+from os.path import join as opj
 from os.path import (
-    abspath,
     lexists,
     relpath,
 )
-from os.path import join as opj
 
 import pytest
 
diff --git a/datalad/distribution/tests/test_siblings.py b/datalad/distribution/tests/test_siblings.py
index ab82c64e6..bddeee536 100644
--- a/datalad/distribution/tests/test_siblings.py
+++ b/datalad/distribution/tests/test_siblings.py
@@ -19,12 +19,12 @@ from datalad.api import (
 )
 from datalad.support.exceptions import InsufficientArgumentsError
 from datalad.support.gitrepo import GitRepo
+from datalad.support.path import basename
+from datalad.support.path import join as opj
 from datalad.support.path import (
-    basename,
     normpath,
     relpath,
 )
-from datalad.support.path import join as opj
 from datalad.tests.utils_pytest import (
     DEFAULT_BRANCH,
     DEFAULT_REMOTE,
diff --git a/datalad/distribution/utils.py b/datalad/distribution/utils.py
index 02d45de08..48838b994 100644
--- a/datalad/distribution/utils.py
+++ b/datalad/distribution/utils.py
@@ -12,8 +12,9 @@
 
 import logging
 import posixpath
-from os.path import isabs, normpath
+from os.path import isabs
 from os.path import join as opj
+from os.path import normpath
 
 from datalad.log import log_progress
 from datalad.support.annexrepo import AnnexRepo
diff --git a/datalad/interface/results.py b/datalad/interface/results.py
index 15859b0b8..a86129ae2 100644
--- a/datalad/interface/results.py
+++ b/datalad/interface/results.py
@@ -22,10 +22,12 @@ from collections.abc import (
 from os.path import (
     isabs,
     isdir,
+)
+from os.path import join as opj
+from os.path import (
     normpath,
     relpath,
 )
-from os.path import join as opj
 from typing import (
     Any,
     Optional,
diff --git a/datalad/interface/utils.py b/datalad/interface/utils.py
index 1b593b81d..90ecf0a0a 100644
--- a/datalad/interface/utils.py
+++ b/datalad/interface/utils.py
@@ -16,12 +16,12 @@ import json
 import logging
 import sys
 from os import listdir
+from os.path import isdir
+from os.path import join as opj
 from os.path import (
-    isdir,
     relpath,
     sep,
 )
-from os.path import join as opj
 from time import time
 from typing import TypeVar
 
@@ -31,7 +31,6 @@ from datalad.dochelpers import single_or_plural
 from datalad.support.exceptions import CapturedException
 from datalad.support.gitrepo import GitRepo
 from datalad.ui import ui
-
 # avoid import from API to not get into circular imports
 from datalad.utils import (
     ensure_unicode,
@@ -39,7 +38,9 @@ from datalad.utils import (
     path_is_subpath,
     path_startswith,
 )
-from datalad.utils import with_pathsep as _with_sep  # TODO: RF whenever merge conflict is not upon us
+from datalad.utils import \
+    with_pathsep as \
+    _with_sep  # TODO: RF whenever merge conflict is not upon us
 
 anInterface = TypeVar('anInterface', bound='Interface')
 
diff --git a/datalad/local/add_archive_content.py b/datalad/local/add_archive_content.py
index 121cb268d..6edad9eff 100644
--- a/datalad/local/add_archive_content.py
+++ b/datalad/local/add_archive_content.py
@@ -21,9 +21,9 @@ from os.path import (
     basename,
     curdir,
     exists,
-    lexists,
 )
 from os.path import join as opj
+from os.path import lexists
 from os.path import sep as opsep
 
 from datalad.consts import ARCHIVES_SPECIAL_REMOTE
diff --git a/datalad/local/foreach_dataset.py b/datalad/local/foreach_dataset.py
index d00f3e04f..50d5e7d4e 100644
--- a/datalad/local/foreach_dataset.py
+++ b/datalad/local/foreach_dataset.py
@@ -56,14 +56,14 @@ from datalad.support.parallel import (
     no_subds_in_futures,
 )
 from datalad.support.param import Parameter
+from datalad.utils import SequenceFormatter
+from datalad.utils import chpwd as chpwd_cm
 from datalad.utils import (
-    SequenceFormatter,
     getpwd,
     nothing_cm,
     shortened_repr,
     swallow_outputs,
 )
-from datalad.utils import chpwd as chpwd_cm
 
 lgr = logging.getLogger('datalad.local.foreach_dataset')
 
diff --git a/datalad/local/tests/test_add_archive_content.py b/datalad/local/tests/test_add_archive_content.py
index 51a799966..bd0697a65 100644
--- a/datalad/local/tests/test_add_archive_content.py
+++ b/datalad/local/tests/test_add_archive_content.py
@@ -19,10 +19,12 @@ from os import unlink
 from os.path import (
     basename,
     exists,
+)
+from os.path import join as opj
+from os.path import (
     lexists,
     pardir,
 )
-from os.path import join as opj
 from pathlib import Path
 
 from datalad.api import (
diff --git a/datalad/support/annexrepo.py b/datalad/support/annexrepo.py
index 3fb6c1524..88e2d52a6 100644
--- a/datalad/support/annexrepo.py
+++ b/datalad/support/annexrepo.py
@@ -24,10 +24,12 @@ from os.path import (
     curdir,
     exists,
     isdir,
+)
+from os.path import join as opj
+from os.path import (
     lexists,
     normpath,
 )
-from os.path import join as opj
 from typing import Dict
 from weakref import (
     WeakValueDictionary,
@@ -43,7 +45,6 @@ from datalad.cmd import (  # KillOutput,
     WitlessProtocol,
 )
 from datalad.consts import WEB_SPECIAL_REMOTE_UUID
-
 # imports from same module:
 from datalad.dataset.repo import RepoInterface
 from datalad.dochelpers import (
diff --git a/datalad/support/archives.py b/datalad/support/archives.py
index 907e47e9c..8e86d87c8 100644
--- a/datalad/support/archives.py
+++ b/datalad/support/archives.py
@@ -27,11 +27,13 @@ from datalad.support.path import (
     exists,
     isabs,
     isdir,
+)
+from datalad.support.path import join as opj
+from datalad.support.path import (
     normpath,
     pardir,
     relpath,
 )
-from datalad.support.path import join as opj
 from datalad.support.path import sep as opsep
 from datalad.utils import (
     Path,
@@ -48,13 +50,13 @@ from datalad.utils import (
 if (cfg.obtain('datalad.runtime.use-patool', default=False,
                valtype=anything2bool)
         or not external_versions['cmd:7z']):
-    from datalad.support.archive_utils_patool import (
-        decompress_file as _decompress_file,  # other code expects this to be here
-    )
+    from datalad.support.archive_utils_patool import \
+        decompress_file as \
+        _decompress_file  # other code expects this to be here
 else:
-    from datalad.support.archive_utils_7z import (
-        decompress_file as _decompress_file,  # other code expects this to be here
-    )
+    from datalad.support.archive_utils_7z import \
+        decompress_file as \
+        _decompress_file  # other code expects this to be here
 
 lgr = logging.getLogger('datalad.support.archives')
 
diff --git a/datalad/support/due_utils.py b/datalad/support/due_utils.py
index 5c95abd22..cedee850f 100644
--- a/datalad/support/due_utils.py
+++ b/datalad/support/due_utils.py
@@ -12,7 +12,6 @@ from ..utils import (
     never_fail,
     swallow_logs,
 )
-
 # Note Text was added/exposed only since DueCredit 0.6.5
 from .due import (
     Doi,
diff --git a/datalad/support/gitrepo.py b/datalad/support/gitrepo.py
index 594b9a37e..8870a1977 100644
--- a/datalad/support/gitrepo.py
+++ b/datalad/support/gitrepo.py
@@ -36,11 +36,13 @@ from os.path import (
     dirname,
     exists,
     isabs,
+)
+from os.path import join as opj
+from os.path import (
     pardir,
     relpath,
     sep,
 )
-from os.path import join as opj
 from re import Pattern
 from typing import (
     TYPE_CHECKING,
@@ -110,7 +112,6 @@ from .exceptions import (
     InvalidGitRepositoryError,
     NoSuchPathError,
 )
-
 # imports from same module:
 from .external_versions import external_versions
 from .network import (
diff --git a/datalad/support/locking.py b/datalad/support/locking.py
index 22185faaa..aece9c329 100644
--- a/datalad/support/locking.py
+++ b/datalad/support/locking.py
@@ -1,9 +1,7 @@
 import logging
 from contextlib import contextmanager
 
-from fasteners import (
-    InterProcessLock,
-)
+from fasteners import InterProcessLock
 
 from datalad.support.exceptions import CapturedException
 
diff --git a/datalad/support/network.py b/datalad/support/network.py
index acaba93b5..17f4c1653 100644
--- a/datalad/support/network.py
+++ b/datalad/support/network.py
@@ -28,15 +28,17 @@ from urllib.error import URLError
 from urllib.parse import (
     ParseResult,
     parse_qsl,
-    quote,
+)
+from urllib.parse import quote
+from urllib.parse import quote as urlquote
+from urllib.parse import unquote as urlunquote
+from urllib.parse import (
     urlencode,
     urljoin,
     urlparse,
     urlsplit,
     urlunparse,
 )
-from urllib.parse import quote as urlquote
-from urllib.parse import unquote as urlunquote
 from urllib.request import (
     Request,
     url2pathname,
diff --git a/datalad/support/path.py b/datalad/support/path.py
index 46b15291d..d5760c2d9 100644
--- a/datalad/support/path.py
+++ b/datalad/support/path.py
@@ -15,7 +15,6 @@ from __future__ import annotations
 # TODO: RF and move all paths related functions from datalad.utils in here
 import os
 import os.path as op
-
 # to not pollute API importing as _
 from collections import defaultdict as _defaultdict
 from collections.abc import (
diff --git a/datalad/support/sshconnector.py b/datalad/support/sshconnector.py
index fc38b39d8..0d059325a 100644
--- a/datalad/support/sshconnector.py
+++ b/datalad/support/sshconnector.py
@@ -34,7 +34,6 @@ from datalad.support.exceptions import (
     ConnectionOpenFailedError,
 )
 from datalad.support.external_versions import external_versions
-
 # importing the quote function here so it can always be imported from this
 # module
 # this used to be shlex.quote(), but is now a cross-platform helper
diff --git a/datalad/support/tests/test_annexrepo.py b/datalad/support/tests/test_annexrepo.py
index b9ec78ba8..913fc3de4 100644
--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -23,10 +23,12 @@ from os import mkdir
 from os.path import (
     curdir,
     exists,
+)
+from os.path import join as opj
+from os.path import (
     pardir,
     relpath,
 )
-from os.path import join as opj
 from queue import Queue
 from shutil import copyfile
 from unittest.mock import patch
@@ -49,7 +51,6 @@ from datalad.consts import (
 from datalad.distribution.dataset import Dataset
 from datalad.runner.gitrunner import GitWitlessRunner
 from datalad.support import path as op
-
 # imports from same module:
 from datalad.support.annexrepo import (
     AnnexJsonProtocol,
@@ -79,6 +80,9 @@ from datalad.tests.utils_pytest import (
     OBSCURE_FILENAME,
     SkipTest,
     assert_cwd_unchanged,
+)
+from datalad.tests.utils_pytest import assert_dict_equal as deq_
+from datalad.tests.utils_pytest import (
     assert_equal,
     assert_false,
     assert_in,
@@ -121,7 +125,6 @@ from datalad.tests.utils_pytest import (
     with_tree,
     xfail_buggy_annex_info,
 )
-from datalad.tests.utils_pytest import assert_dict_equal as deq_
 from datalad.utils import (
     Path,
     chpwd,
diff --git a/datalad/support/tests/test_due_utils.py b/datalad/support/tests/test_due_utils.py
index fa8755490..5522447f9 100644
--- a/datalad/support/tests/test_due_utils.py
+++ b/datalad/support/tests/test_due_utils.py
@@ -7,9 +7,7 @@ from ...tests.utils_pytest import (
     swallow_logs,
     with_tempfile,
 )
-from ..due import (
-    due,
-)
+from ..due import due
 from ..due_utils import duecredit_dataset
 from ..external_versions import external_versions
 
diff --git a/datalad/support/tests/test_parallel.py b/datalad/support/tests/test_parallel.py
index 48ee6f11d..3d6813643 100644
--- a/datalad/support/tests/test_parallel.py
+++ b/datalad/support/tests/test_parallel.py
@@ -21,7 +21,6 @@ import pytest
 from datalad import lgr
 from datalad.support import path as op
 from datalad.support.exceptions import IncompleteResultsError
-
 # absolute import only to be able to run test without `nose` so to see progress bar
 from datalad.support.parallel import (
     ProducerConsumer,
diff --git a/datalad/support/vcr_.py b/datalad/support/vcr_.py
index 8233cbb3c..413a53866 100644
--- a/datalad/support/vcr_.py
+++ b/datalad/support/vcr_.py
@@ -31,8 +31,10 @@ try:
     # https://github.com/kevin1024/vcrpy/issues/215
     import vcr.patch as _vcrp
     try:
-        from requests.packages.urllib3.connectionpool import HTTPConnection as _a
-        from requests.packages.urllib3.connectionpool import VerifiedHTTPSConnection as _b
+        from requests.packages.urllib3.connectionpool import \
+            HTTPConnection as _a
+        from requests.packages.urllib3.connectionpool import \
+            VerifiedHTTPSConnection as _b
     except ImportError:
         def returnnothing(*args, **kwargs):
             return()
diff --git a/datalad/tests/test_utils.py b/datalad/tests/test_utils.py
index 4800abe38..03d22fdaf 100644
--- a/datalad/tests/test_utils.py
+++ b/datalad/tests/test_utils.py
@@ -28,10 +28,12 @@ from os.path import (
     expanduser,
     expandvars,
     isabs,
+)
+from os.path import join as opj
+from os.path import (
     normpath,
     pardir,
 )
-from os.path import join as opj
 from unittest.mock import patch
 
 import pytest
diff --git a/datalad/tests/utils_pytest.py b/datalad/tests/utils_pytest.py
index ba941701d..6f4b7eec9 100644
--- a/datalad/tests/utils_pytest.py
+++ b/datalad/tests/utils_pytest.py
@@ -41,7 +41,6 @@ from datalad.support.external_versions import (
 from datalad.support.keyring_ import MemoryKeyring
 from datalad.support.network import RI
 from datalad.support.vcr_ import *
-
 # TODO this must go
 from datalad.utils import *
 
@@ -355,9 +354,7 @@ def skip_nomultiplex_ssh(func):
 import os
 
 from datalad.distribution.dataset import Dataset
-from datalad.support.annexrepo import (
-    AnnexRepo,
-)
+from datalad.support.annexrepo import AnnexRepo
 from datalad.support.gitrepo import GitRepo
 
 from ..utils import (
diff --git a/datalad/typing.py b/datalad/typing.py
index 8f4b6cd64..5c1dd9977 100644
--- a/datalad/typing.py
+++ b/datalad/typing.py
@@ -16,22 +16,14 @@ else:
     pass
 
 if sys.version_info >= (3, 10):
-    from typing import (
-        ParamSpec,
-    )
+    from typing import ParamSpec
 else:
-    from typing_extensions import (
-        ParamSpec,
-    )
+    from typing_extensions import ParamSpec
 
 if sys.version_info >= (3, 8):
-    from typing import (
-        Literal,
-    )
+    from typing import Literal
 else:
-    from typing_extensions import (
-        Literal,
-    )
+    from typing_extensions import Literal
 
 __all__ = ["Literal", "ParamSpec", "T", "K", "V", "P"]
 
diff --git a/datalad/ui/dialog.py b/datalad/ui/dialog.py
index 618ea1f88..b65e0c1c4 100644
--- a/datalad/ui/dialog.py
+++ b/datalad/ui/dialog.py
@@ -22,7 +22,6 @@ import getpass
 import os
 import sys
 import time
-
 #!!! OPT adds >100ms to import time!!!
 # from unittest.mock import patch
 from collections import deque
diff --git a/datalad/utils.py b/datalad/utils.py
index eeef97e7a..5f6d340dd 100644
--- a/datalad/utils.py
+++ b/datalad/utils.py
@@ -41,7 +41,6 @@ from functools import (
     wraps,
 )
 from itertools import tee
-
 # this import is required because other modules import opj from here.
 from os.path import (
     abspath,
@@ -63,9 +62,7 @@ from os.path import (
     split,
     splitdrive,
 )
-from pathlib import (
-    Path,
-)
+from pathlib import Path
 from shlex import quote as shlex_quote
 from shlex import split as shlex_split
 from tempfile import NamedTemporaryFile
```
</details>

so major diff also that with isort we do not do a "single import in a tuple" which sounds like the right thing to do (and hatch also does not do them for some but does for others? unclear to me)

Reply:
actually those "single import as tuple" are just because hatch did not bother to make it non-tuple after removing other imports from it... so it is not trying to bring them back if `isort`ed . But there is some difference in how `hatch fmt` orders imports and then it likes adding empty line separations.

Overall divergences quite small, and may be we could eventually arrive to 1 single choice of the tool